hw2/1_mini_logo.hs

```diff
@@ -1 +1,89 @@
 -- Mini Logo
+
+-- Mini Logo is a simplified version of the Logo language for programming
+-- 2D graphics. The idea behind Logo and Mini Logo is to describe simple
+-- line graphics through commands to move a pen from one position to
+-- another. The pen can either be "up" or "down". Positions are given
+-- by pairs of integers. Functions can be defined (using def) and called
+-- (using call) to reuse groups of commands. The syntax of Mini Logo is as
+-- follows
+
+-- part a: define the abstract syntax for mini logo as Cmd
+
+data Mode = Up | Down deriving (Show)
+
+data Coord = Ints (Int, Int) | Names (String, String) deriving (Show)
+
+data Pars = Pars String Pars | Name String deriving (Show)
+
+data Vals = Vals Int Vals | Int deriving (Show)
+
+data Cmd
+  = Pen Mode
+  | MoveTo Coord
+  | Def String Pars Cmd
+  | Call Vals String
+  | Seq Cmd Cmd
+  deriving (Show)
+
+seqCmds :: [Cmd] -> Cmd
+seqCmds [cmd] = cmd
+seqCmds (cmd : cmds) = Seq cmd (seqCmds cmds)
+
+-- part b: Write Mini Logo function `vector`
+-- def vector (x1, y1, x2, y2) pen up ; moveto (x1, y1);
+-- pen down; moveto (x2, y2); pen up
+
+vector :: Cmd
+vector =
+  Def
+    "vector"
+    (Pars "x1" (Pars "y1" (Pars "x2" (Name "y2"))))
+    ( seqCmds
+        [ Pen Up,
+          MoveTo (Names ("x1", "y1")),
+          Pen Down,
+          MoveTo (Names ("x2", "y2")),
+          Pen Up
+        ]
+    )
+
+-- part c: define a haskell function steps :: Int -> Cmd
+-- that constructs a Mini Logo program to draw a stair
+-- on n steps.
+
+moveTo x y = MoveTo (Ints (x, y))
+
+step :: Int -> Int -> Cmd
+step x y
+  | x > 1 =
```

Comment:
It's valid to pattern match on constants, so if we want to, we can get rid of the conditions. I think it's a matter of opinion which you think is better, but I think it reads a little cleaner. 
![image](https://github.com/osu-mp/cs581/assets/79019391/6b06fd66-b60c-4b37-bbb4-74cbffb220a8)

Reply:
I think your version looks better, I'll switch that up